src/utils/config.ts

```diff
@@ -0,0 +1,464 @@
+  }
+
+  const configPath = path.join(buildSourceFolder, 'sasjsconfig.json')
+
+  await createFile(configPath, JSON.stringify(config, null, 2))
+
+  return configPath
+}
+
+export async function getFolders() {
+  const configPath = '../config.json'
+  const config = await readFile(path.join(__dirname, configPath))
+  if (config) {
+    const configJson = JSON.parse(config)
+    return Promise.resolve(configJson.folders)
+  }
+  return Promise.reject()
+}
+
+export async function getSourcePaths(buildSourceFolder: string) {
+  let configuration = await getConfiguration(
+    path.join(buildSourceFolder, 'sasjsconfig.json')
+  )
+  if (!configuration) {
+    configuration = { macroFolders: [] }
+  }
+
+  const sourcePaths = configuration.macroFolders
+    ? configuration.macroFolders.map((macroPath: string) =>
+        path.join(buildSourceFolder, macroPath)
+      )
+    : []
+  const macroCorePath = path.join(
+    process.projectDir,
+    'node_modules',
+    '@sasjs',
+    'core'
+  )
+  sourcePaths.push(macroCorePath)
+
+  return sourcePaths
+}
+
+/**
+ * Returns SAS program folders from configuration.
+ * This list includes both common and target-specific folders.
+ * @param {string} targetName - name of the configuration.
+ */
+export async function getProgramFolders(targetName: string) {
+  let programFolders: string[] = []
+  const projectRoot = await getProjectRoot()
+  const localConfig = await getConfiguration(
+    path.join(projectRoot, 'sasjs', 'sasjsconfig.json')
+  ).catch(() => null)
+  if (localConfig && localConfig.programFolders) {
+    programFolders = programFolders.concat(localConfig.programFolders)
+  }
+
+  const { target } = await findTargetInConfiguration(targetName)
+
+  if (!target) {
+    throw new Error(
+      'Target not found.\nPlease check the target name and try again, or use `sasjs add` to add a new target.'
+    )
+  }
+
+  if (target.programFolders) {
+    programFolders = programFolders.concat(target.programFolders)
+  }
+
+  if (!programFolders.length) {
+    console.log(
+      chalk.yellowBright(
+        'No program folders found. If you have SAS program dependencies, please specify the program paths in the `programFolders` array in your configuration.'
+      )
+    )
+  }
+  return programFolders
+}
+
+/**
+ * Returns SAS macro folders from configuration.
+ * This list includes both common and target-specific folders.
+ * @param {string} targetName - name of the configuration.
+ */
+export async function getMacroFolders(targetName: string) {
+  let macroFolders: string[] = []
+  const projectRoot = await getProjectRoot()
+  const localConfig = await getConfiguration(
+    path.join(projectRoot, 'sasjs', 'sasjsconfig.json')
+  ).catch(() => null)
+  if (localConfig && localConfig.programFolders) {
+    macroFolders = macroFolders.concat(localConfig.programFolders)
+  }
+
+  const { target } = await findTargetInConfiguration(targetName)
+
+  if (!target) {
+    throw new Error(
```

Comment:
this piece of code is unreachable (and redundant as code throws in `findTargetInConfiguration` if there's no target.

Reply:
I've removed that bit of code.